### PR TITLE
config.fish: improve documentation on creating Fish functions

### LINKED ISSUE #128: There is no && operator in fish
There is no && operator in fish, so cd_mkdir and other rules using chaining does not work. 
";and" should be used in fish instead of &&.
http://superuser.com/questions/446930/chain-fish-commands-via-or
screenshot from 2015-04-25 12 06 58

### PR DESCRIPTION
**Pull request by scorphus**
How about this as solution to #128? Please review and comment. Thanks!

### CODE REVIEW

**Comment by krendelkoph** [comment]
This solution works great, thanks.

**Comment by scorphus** [comment]
It depends on `sed`, though. But I believe it's certain to be available.

**Comment by nvbn** [comment]
Can it also include [solution for stdin](https://github.com/nvbn/thefuck/issues/76#issuecomment-96103809)?

**Comment by scorphus** [comment]
Let me try that, @nvbn

**Comment by scorphus** [comment]
@nvbn do you mean adopting the workaround as mentioned in https://github.com/nvbn/thefuck/issues/76#issuecomment-96103809?

**Comment by scorphus** [comment]
BTW, I don't think it works, although I'm still trying.

**Comment by nvbn** [comment]
@scorphus yep.

**Comment by scorphus** [comment]
@nvbn nice, 4bce525 does just that. Let's wait and see if anyone confirms it solves #76.

**Comment by krendelkoph** [comment]
@scorphus, it solves both bugs for me.

File: README.md [hunk 1]
```diff
@@ -117,8 +117,16 @@ alias FUCK='fuck'
...
 ```fish
-function fuck
-    eval (thefuck $history[1])
+function __thefuck_repl --description 'Replace operators into fish-compatible'
+    set -l tmp (echo $argv | sed 's/ && / ; and /g')
+    echo $tmp | sed 's/ || / ; or /g'
+end
+
+function fuck --description 'Correct your previous console command'
+    set -l eval_script /tmp/.thefuck.fish
```
Comment (daenney) [inline README.md]: This should probably use `$TMPDIR` instead / when it is set.

Comment (brassg) [inline README.md]: Hey how is this work I want to be a private socialism

Comment (scorphus) [inline README.md]: Absolutely, @daenney! I'll update this pull request ASAP. Thank you!
Em 26/04/2015 06:20, "Daniele Sluijters" <notifications@github.com>
escreveu:

Comment (kingwood2112) [inline README.md]: Wsup

**Comment by daenney** [comment]
@scorphus One thing on the `$TMPDIR` thing, a lot of Linux distributions don't appear to have it set by default. OS X does but not every Linux distro does so there should be a fallback there to `/tmp` if it's not set.

**Comment by scorphus** [comment]
Yeah, absolutely. First I read http://en.wikipedia.org/wiki/TMPDIR and changed it. But then I thought it over and decided to `vagrant up` Ubuntu 14.04, CentOS 6.4 and Debian 7 to test it and, no surprise. Now I'm testing with `mktemp` and it's working just fine in all cases.

**Comment by scorphus** [comment]
Ouch, no... too bad. `mktemp`'s implementations differ amongst Linux and OS X

**Comment by scorphus** [comment]
So, how about this:

```fish
function __thefuck_repl --description 'Replace operators into fish-compatible'
    set -l tmp (echo $argv | sed 's/ && / ; and /g')
    echo $tmp | sed 's/ || / ; or /g'
end

function fuck --description 'Correct your previous console command'
    set -l eval_script (mktemp 2>/dev/null ; or mktemp -t 'thefuck')
    thefuck $history[1] > $eval_script
    eval (__thefuck_repl (cat $eval_script))
    rm $eval_script
end
```

**Comment by nvbn** [comment]
Can I merge it? Or it's not a final solution?

**Comment by nvbn** [comment]
I moved aliases to [wiki](https://github.com/nvbn/thefuck/wiki/Shell-aliases), it will be faster to change them there.

**Comment by scorphus** [comment]
I think it's good to go. Ping, @daenney, what do you think?
Em 27/04/2015 02:58, "Vladimir Iakovlev" <notifications@github.com>
escreveu:

**Comment by daenney** [comment]
Ya, I think this should work. Needs a rebase though since Github thinks it has a merge conflict.

**Comment by scorphus** [comment]
Thanks for the input, @daenney!

The part of the README.md file regarding shell aliases and/or functions was move over to the wiki, that's why this is conflicting with master. I'll move the solution to the wiki too.

**Comment by scorphus** [comment]
One thing to remind ourselves of, though, is that from now on, every single rule that involves logical operators (`&&` and `||` for instance) should enclose them in blank spaces, like the following: `cmd_x && cmd_y` or `“cmd_x || cmd_z”`.

File: README.md (hunk 1 above)
Comment (kingwood2112) [inline README.md]: Hi,

I'd like to add you to my professional network on LinkedIn.

- R

Confirm that you know R: https://www.linkedin.com/e/v2?e=-vzcnar-i99jtls3-2p&amp;a=preRegInvite&amp;tracking=eml-guest-invite-cta&amp;ek=invite_guest&amp;sharedKey=s6w-pACt&amp;invitationID=6000889217104773120


You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-vzcnar-i99jtls3-2p&amp;t=lun&amp;midToken=AQFmrFVhGEd36Q&amp;ek=invite_guest&amp;loid=AQErCKscOMsdpAAAAU0dxCmGDljlAbEvVjwoZVCVd8sfXxIrRpV4LVpMT4upVN0b8JoV0xD_8RgmMF1SrVZYCwuJGydNEJpUbM4-t8bkthh1UE1ivDFkeaWLUMizUiZbIQZSTyckiSi0kV6O0a2-J-s9aPiCzeUJhLVdqNsvenv_YktBLHXAyeJQFfoqaMVBcoh5&amp;eid=-vzcnar-i99jtls3-2p Learn why we included this at the following link: https://www.linkedin.com/e/v2?e=-vzcnar-i99jtls3-2p&a=customerServiceUrl&ek=invite_guest&articleId=4788
&copy; 2015, LinkedIn Corporation. 2029 Stierlin Ct. Mountain View, CA 94043, USA

**Pull request closed without merging**